chapters/en/slides/module4/module4_01.Rmd

```diff
@@ -0,0 +1,345 @@
+
+While the facetted plot makes it easy to answer the question:
+"Which is the most common movie genre within each MPAA rating group?",
+it is harder to answer the question:
+"Which is the most common MPAA rating assigned to movies of a particular genre?".
+
+To better answer the second question,
+we could have switched which column we facetted by and which we have on the y-axis,
+but there is no great solution for getting the best of both worlds in a faceted chart.
+
+---
+
+## Heatmaps are effective for visualizing counts of two dimensional categorical data
+
+```{python}
+alt.Chart(movies_extended).mark_rect().encode(
+    alt.Color('count()'),
+    alt.X('MPAA Rating'),
+    alt.Y('Major Genre', sort='color'))
+```
+
+Notes:
+
+To be able to effectively answer both the questions from the previous slide
+with a single visualization,
+we can represent the count with color
+instead of mapping it to one of the axes.
+
+Using `mark_rect`,
+we can now create a heatmap where 
+the two categorical are mapped to one axis each
+and we don't need to facet.
+Sorting on color/count puts the genres with many observations close together,
+similar to how we sorted on `'x'` and `'y'` in previous modules.
+
+If we want to compare which genre is most common for a certain rating,
+we compare the colors in the same *column*.
+If we instead are interested in the most common rating assigned to a movie
+we compare the columns *row-wise*.
+We can quickly see that most dramas are rated PG-13 or R
+and most horror movies are rated R.
+
+This visualization is effective
+for quickly communicating the main takeaways from the two questions
+and giving us an overview of the data,
+but it is harder  to tell that exact count for each color
+so if that is of great importance a bar chart is more suitable.
+
+As you can see the default colorscale is different from the marks we have used so far.
+We will talk more about colorscale in a later module,
+but briefly the change in hue from one color to another
+allows for more resolution in being able to tell
+which color corresponds to a certain value.
+
+For the same reasons a density plot was not an appropriate replacement of the categorical barchart,
+we cannot replace this categorical heatmap with a 2D denity plot.
+
+---
+
+## Using both the color and marker size to indicate the count creates a more effective visualization.
+
+```{python}
+# Sort since according to the natural order of the ratings
+mpaa_ordered = ['G', 'PG', 'PG-13', 'R']
+alt.Chart(movies_extended).mark_circle().encode(
+    alt.Color('count()'),
+    alt.Size('count()'),
+    alt.X('MPAA Rating', sort=mpaa_ordered),
+    alt.Y('Major Genre', sort='x'))
+```
+
+Notes:
+TODO an alternative strategy is to not teach mark_rect for categorical and go straight to this plot... I like mentioning it to contrast it with this, but not sure if there would be confusion or if the heatmap for some reason implies continuous data.
```

Comment:
I think what you have (if you add in stacked bar charts) is fine. Let's remove this.